exovetter/vetters.py

```diff
@@ -132,16 +119,122 @@ def run(self, tce, lightcurve):
             'norm_lpp': self.norm_lpp,
             'plot_data': self.plot_data}
 
-    def plot(self):
+    def plot(self):  # pragma: no cover
         if self.plot_data is None:
             raise ValueError(
                 'LPP plot data is empty. Execute self.run(...) first.')
 
         # target is populated in TCE, assume it already exists.
-        target = self.tce['target_name']
+        target = self.tce.get('target_name', 'Target')
         lpp.plot_lpp_diagnostic(self.plot_data, target, self.norm_lpp)
 
 
+class Sweet(BaseVetter):
+    """Class to handle SWEET Vetter functionality.
+
+    Parameters
+    ----------
+    threshold_sigma : float
+        Threshold for comparing signal to transit period.
+
+    Attributes
+    ----------
+    tce, lc
+        Inputs to :meth:`run`.
+
+    result : dict
+        ``'amp'`` contains the best fit amplitude, its uncertainty, and
+        amplitude-to-uncertainty ratio for half-period, period, and
+        twice the period. ``'msg'`` contains warnings, if applicable.
+        Populated by :meth:`run`.
+
+    lsf : `~exovetter.utils.WqedLSF`
+        Least squares fit object, populated by :meth:`run`.
+
+    """
+    def __init__(self, threshold_sigma=3):
+        self.tce = None
+        self.lc = None
+        self.result = None
+        self.lsf = None
+        self.threshold_sigma = threshold_sigma
+
+    def run(self, tce, lightcurve):
+        self.tce = tce
+        self.lc = lightcurve
+        epoch = tce.get_epoch(getattr(exo_const, lightcurve.time_format))
+
+        # TODO: Do we want results to have unit? If so, what?
```

Comment:
Not clear to me if you want the results to be `Quantity` or not. If yes, what are the units? Would be nice if result is naturally `Quantity` if we can make the low level functions take `Quantity` but that is significant refactoring work. See #16.

Reply:
This run method understands what is being passed to it (a lightkurve object), and so is the place to deal with issues of Quantities. 

a) If you choose to implement _do_fit as a method that accepts quantities, run() is responsible for ensuring that the values passed are indeed quantities.

b) If you choose to implement _do_fit as a function that expects numpy arrays, then run() should ensure that the arrays passed are of the correct units.

In our specific case here, the implementation returns the amplitude of the out of transit variation in the same units (e.g ppm) as the input lightcurve. it looks like lightkurve stores a numpy array for flux, and has a separate atribute for Unit. In this case, what I would likely do is pass numpy arrays to _do_fit(), then assign units to the outputs in run(). So something like

```
def run(self):
    ...
   flux = lightcurve.flux_quantity
   ...
   result, lsf = ...
    result[:,0] = u.Quantity(result[:,0], lightcurve.flux_unix)
    result[:,1] = u.Quantity(result[:,0], lightcurve.flux_unix)
```

Does this comment resolve all the questions?